include/tvm/ir/module.h

```diff
@@ -349,6 +393,9 @@ class IRModuleNode : public Object {
    */
   std::unordered_set<String> import_set_;
   friend class IRModule;
+
+ public:
+  void ExtractPrimFuncConstants(tir::PrimFunc func);
```

Comment:
Just a suggestion : Since this is part of the unified IR module definition, should we keep it for BaseFunc for now ? and implement it for tir::PrimFunc for this PR. Thus, the community might able to do a similiar for relay::Function to complete the implementation. This way we might able to remove the forward declarations as well.

Reply:
Yeah I wanted to second @manupa-arm here :-) Usually we don't add these specific methods attached to IRModule

Reply:
Altered method.